### PR TITLE
LPS-117306 | Segment criteria for custom fields of type number (decimal or integer) cannot be compared as numbers

### PR DESCRIPTION
**Pull request by ricardocousolr**
Hi @liferay-tango,

This is pr to solve issue [LPS-117306](https://issues.liferay.com/browse/LPS-117306). The solution is very much analogous to that for [LPS-87867](https://issues.liferay.com/browse/LPS-87867). 

Notice, though, that I have associated:
- `ExpandoColumnConstants.SHORT`
- `ExpandoColumnConstants.INTEGER`
- `ExpandoColumnConstants.LONG`

with `IntegerEntityField`.

I've also associated:
- `ExpandoColumnConstants.DOUBLE`
- `ExpandoColumnConstants.FLOAT`

with `DoubleEntityField`.

Please let me know if you have any questions.

Thanks!

### CODE REVIEW

**Comment by darquesdev** [comment]
ci:test:sf

**Comment by darquesdev** [comment]
ci:test:relevant

**Comment by darquesdev** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/model/listener/UserExpandoColumnModelListener.java [hunk 1]
```diff
@@ -191,6 +192,13 @@ else if (expandoColumn.getType() == ExpandoColumnConstants.DATE) {
 				locale -> Field.getSortableFieldName(encodedIndexedFieldName),
 				locale -> encodedIndexedFieldName);
 		}
+		else if ((expandoColumn.getType() == ExpandoColumnConstants.SHORT) ||
```
Comment (darquesdev) [inline modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/model/listener/UserExpandoColumnModelListener.java:202]: Normally It's good to sort conditions alphabetically, if logic is not affected with the change

```java
		else if ((expandoColumn.getType() == ExpandoColumnConstants.INTEGER) ||
				 (expandoColumn.getType() == ExpandoColumnConstants.LONG) ||
				 (expandoColumn.getType() == ExpandoColumnConstants.SHORT)) {

			entityField = new IntegerEntityField(
				encodedName, locale -> encodedIndexedFieldName);
		}
```

File: modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/model/listener/OrganizationExpandoColumnModelListener.java [hunk 2]
```diff
@@ -185,6 +186,13 @@ else if (expandoColumn.getType() == ExpandoColumnConstants.DATE) {
 				locale -> Field.getSortableFieldName(encodedIndexedFieldName),
 				locale -> encodedIndexedFieldName);
 		}
+		else if ((expandoColumn.getType() == ExpandoColumnConstants.SHORT) ||
```
Comment (darquesdev) [inline modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/model/listener/OrganizationExpandoColumnModelListener.java:196]: Normally It's good to sort conditions alphabetically, if logic is not affected with the change

File: modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/odata/retriever/test/UserODataRetrieverCustomFieldsTest.java [hunk 3]
```diff
@@ -241,6 +241,76 @@ public void testGetUsersFilterByCustomFieldWithEqualsAndDateTextType()
 		Assert.assertEquals(user1, users.get(0));
 	}
 
+	@Test
+	public void testGetUsersFilterByCustomFieldWithEqualsAndIntegerKeywordType()
```
Comment (darquesdev) [inline modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/odata/retriever/test/UserODataRetrieverCustomFieldsTest.java:315]: I'd add a test for `ExpandoColumnConstants.SHORT` and another for `ExpandoColumnConstants.LONG`, does it make sense for you?

File: modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/odata/retriever/test/UserODataRetrieverCustomFieldsTest.java [hunk 4]
```diff
@@ -241,6 +241,76 @@ public void testGetUsersFilterByCustomFieldWithEqualsAndDateTextType()
...
+
+		List<User> users = _getODataRetriever().getResults(
+			TestPropsValues.getCompanyId(), filterString,
+			LocaleUtil.getDefault(), 0, 1);
+
+		Assert.assertEquals(user1, users.get(0));
+	}
+
+	@Test
+	public void testGetUsersFilterByCustomFieldWithEqualsAndIntegerTextType()
```
Comment (darquesdev) [inline modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/odata/retriever/test/UserODataRetrieverCustomFieldsTest.java:350]: I'd add a test for `ExpandoColumnConstants.SHORT` and another for `ExpandoColumnConstants.LONG`, does it make sense for you?

File: modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/odata/retriever/test/OrganizationODataRetrieverCustomFieldsTest.java [hunk 5]
```diff
@@ -253,6 +253,78 @@ public void testGetOrganizationsFilterByCustomFieldWithEqualsAndDateTextType()
 		Assert.assertEquals(organization1, organizations.get(0));
 	}
 
+	@Test
+	public void testGetOrganizationsFilterByCustomFieldWithEqualsAndIntegerKeywordType()
```
Comment (darquesdev) [inline modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/odata/retriever/test/OrganizationODataRetrieverCustomFieldsTest.java:329]: I'd add a test for `ExpandoColumnConstants.SHORT` and another for `ExpandoColumnConstants.LONG`, does it make sense for you?

File: modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/odata/retriever/test/OrganizationODataRetrieverCustomFieldsTest.java [hunk 6]
```diff
@@ -253,6 +253,78 @@ public void testGetOrganizationsFilterByCustomFieldWithEqualsAndDateTextType()
...
+
+		List<Organization> organizations = _getODataRetriever().getResults(
+			TestPropsValues.getCompanyId(), filterString,
+			LocaleUtil.getDefault(), 0, 1);
+
+		Assert.assertEquals(organization1, organizations.get(0));
+	}
+
+	@Test
+	public void testGetOrganizationsFilterByCustomFieldWithEqualsAndIntegerTextType()
```
Comment (darquesdev) [inline modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/odata/retriever/test/OrganizationODataRetrieverCustomFieldsTest.java:365]: I'd add a test for `ExpandoColumnConstants.SHORT` and another for `ExpandoColumnConstants.LONG`, does it make sense for you?

File: modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/odata/retriever/test/UserODataRetrieverCustomFieldsTest.java [hunk 7]
```diff
@@ -241,6 +241,76 @@ public void testGetUsersFilterByCustomFieldWithEqualsAndDateTextType()
 		Assert.assertEquals(user1, users.get(0));
 	}
 
+	@Test
+	public void testGetUsersFilterByCustomFieldWithEqualsAndDoubleKeywordType()
```
Comment (darquesdev) [inline modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/odata/retriever/test/UserODataRetrieverCustomFieldsTest.java:245]: I'd add a test for `ExpandoColumnConstants.FLOAT`, does it make sense?

File: modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/odata/retriever/test/UserODataRetrieverCustomFieldsTest.java [hunk 8]
```diff
@@ -241,6 +241,76 @@ public void testGetUsersFilterByCustomFieldWithEqualsAndDateTextType()
...
+
+		List<User> users = _getODataRetriever().getResults(
+			TestPropsValues.getCompanyId(), filterString,
+			LocaleUtil.getDefault(), 0, 1);
+
+		Assert.assertEquals(user1, users.get(0));
+	}
+
+	@Test
+	public void testGetUsersFilterByCustomFieldWithEqualsAndDoubleTextType()
```
Comment (darquesdev) [inline modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/odata/retriever/test/UserODataRetrieverCustomFieldsTest.java:280]: I'd add a test for `ExpandoColumnConstants.FLOAT`, what do you think?

**Comment by darquesdev** [comment]
ci:test:sf

**Comment by ricardocousolr** [comment]
Hi @darquesdev,

New pr here #37. (Had to fix some commit messages.)

File: modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/model/listener/UserExpandoColumnModelListener.java (hunk 1 above)
Comment (ricardocousolr) [reply modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/model/listener/UserExpandoColumnModelListener.java:202]: Addressed in #37.

File: modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/model/listener/OrganizationExpandoColumnModelListener.java (hunk 2 above)
Comment (ricardocousolr) [reply modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/model/listener/OrganizationExpandoColumnModelListener.java:196]: Addressed in #37.

**Comment by ricardocousolr** [comment]
Hi @darquesdev,

Sent new pr #38. (Some commit messages had typos.)

**Comment by cgoncas** [comment]
Continues in https://github.com/liferay-tango/liferay-portal/pull/38

**Pull request closed without merging**